modules/date/date-parser.c

```diff
@@ -85,16 +85,22 @@ _parse_timestamp_and_deduce_missing_parts(DateParser *self, WallClockTime *wct,
 {
   const gchar *remainder;
 
-  remainder = wall_clock_time_strptime(wct, self->date_format, input);
-  if (!remainder || remainder[0])
-    return FALSE;
+  for (GList *item = self->date_formats; item; item = item->next)
+    {
+      remainder = wall_clock_time_strptime(wct, item->data, input);
 
```

Comment:
I would probably keep this function just to parse a single timestamp and move the loop to a new function that iterates the list.

That would make a good place for the "date-parser message processing started" log message, which originally included the current format and right now only tells the user how many entries there are in the list. That is not very useful imho.